### ISSUE TITLE
2024 wrap up

### ISSUE THREAD

**Issue by AlbertDominguez** [dlmbl/intro_dl#6]
#### Issues/stuff to improve
- Much more time than scheduled (it was OK as the rest of the afternoon was free anyway)
- A lot of time spent on the XOR
- Live loss curves plotting didn't show on the exercise, but in the solutions
- IMO the introduction to `torch` itself was a bit too abrupt by starting with `nn.Module` directly
- Some people were a bit confused by the loss reduction parameter

#### Ideas for next year 
- Introduce `torch` step-by-step from the beginning: `torch.Tensor`, ops on it, etc. then `nn.Module` (layers, combining them, etc.)
- Implement the perceptron in `torch` (instead of `numpy`) as a first `torch`-based task
